### ISSUE TITLE
Combine should support observables

### ISSUE THREAD

**Issue by hahn-kev** [benlau/asyncfuture#1]
I was a bit surprised to find that this doesn't work:
```
Observable<void> o;
auto all = combine() << o;
```
sure you can just call `o.future()` but it seems like that should be overloaded in combine.
Could this just be added to the `Combinator`?
```
    template <typename T>
    Combinator& operator<<(Observable<T> observable)
    {
        combinedFuture->addFuture(observable.future());
        return *this;
    }
```

**Comment by benlau** [comment]
Hi @hahn-kev ,

Should be a good idea. Will implement it when I am free.

Thx.

**Comment by benlau** [comment]
Added in master branch. Will release later

**Comment by benlau** [comment]
It is already released with 0.3.6.1
